src/entity/projectile/Trident.php

```diff
@@ -0,0 +1,180 @@
+	){
+		if($item->isNull()){
+			throw new \InvalidArgumentException("Trident must have a count of at least 1");
+		}
+		parent::__construct($location, $shootingEntity, $nbt);
+	}
+
+	protected function getInitialSizeInfo() : EntitySizeInfo{ return new EntitySizeInfo(0.35, 0.25); }
+
+	protected function getInitialDragMultiplier() : float{ return 0.01; }
+
+	protected function getInitialGravity() : float{ return 0.1; }
+
+	protected function initEntity(CompoundTag $nbt) : void{
+		parent::initEntity($nbt);
+
+		$this->spawnedInCreative = $nbt->getByte(self::TAG_SPAWNED_IN_CREATIVE, 0) === 1;
+	}
+
+	public function saveNBT() : CompoundTag{
+		$nbt = parent::saveNBT();
+		$nbt->setTag(self::TAG_ITEM, $this->item->nbtSerialize());
+		$nbt->setByte(self::TAG_SPAWNED_IN_CREATIVE, $this->spawnedInCreative ? 1 : 0);
+		return $nbt;
+	}
+
+	protected function onFirstUpdate(int $currentTick) : void{
+		$owner = $this->getOwningEntity();
+		$this->spawnedInCreative = $owner instanceof Player && $owner->isCreative();
+
+		parent::onFirstUpdate($currentTick);
+	}
+
+	protected function entityBaseTick(int $tickDiff = 1) : bool{
+		if($this->closed){
+			return false;
+		}
+		//TODO: Loyalty enchantment.
+
+		return parent::entityBaseTick($tickDiff);
+	}
+
+	protected function despawnsOnEntityHit() : bool{
+		return false;
+	}
+
+	protected function onHitEntity(Entity $entityHit, RayTraceResult $hitResult) : void{
+		parent::onHitEntity($entityHit, $hitResult);
+
+		$this->canCollide = false;
+		$this->broadcastSound(new TridentHitEntitySound());
+		$this->setMotion(new Vector3($this->motion->x * -0.01, $this->motion->y * -0.1, $this->motion->z * -0.01));
+	}
+
+	protected function onHitBlock(Block $blockHit, RayTraceResult $hitResult) : void{
+		parent::onHitBlock($blockHit, $hitResult);
+		$this->canCollide = true;
+		$this->broadcastSound(new TridentHitGroundSound());
+	}
+
+	public function getItem() : TridentItem{
+		return clone $this->item;
+	}
+
+	public function setItem(TridentItem $item) : void{
+		if($item->isNull()){
+			throw new \InvalidArgumentException("Trident must have a count of at least 1");
+		}
+		if($this->item->hasEnchantments() !== $item->hasEnchantments()){
+			$this->networkPropertiesDirty = true;
+		}
+		$this->item = clone $item;
+	}
+
+	public function canCollideWith(Entity $entity) : bool{
+		return $this->canCollide && $entity->getId() !== $this->ownerId && parent::canCollideWith($entity);
+	}
+
+	public function onCollideWithPlayer(Player $player) : void{
+		if($this->blockHit !== null){
+			$this->pickup($player);
+		}
+	}
+
+	private function pickup(Player $player) : void{
+		$shouldDespawn = false;
+
+		$playerInventory = $player->getInventory();
+		$ev = new EntityItemPickupEvent($player, $this, $this->getItem(), $playerInventory);
+		if($player->hasFiniteResources() && !$playerInventory->canAddItem($ev->getItem())){
+			$ev->cancel();
+		}
+		if($this->spawnedInCreative){
+			$ev->cancel();
+			$shouldDespawn = true;
```

Comment:
This is going to despawn tridents thrown by creative players if a survival player tries to pick them up, even though no item will be received. Doesn't seem right to me.

Reply:
That's the expected behavior

Reply:
Expected by who?

Reply:
I'm pretty sure this is a bug in Bedrock. No other items behave like this.
Arrows fired in creative aren't expected to despawn when a survival player steps on them.

Also, from the wiki:

> Only the owner of a trident thrown in Creative mode or enchanted with Loyalty can pick it up; other players in Creative and players in Survival mode including its own owner can't pick up the trident thrown in Creative.

I would expect this to work the same way arrows do, with modes NONE/ANY/CREATIVE pickup modes.